### PR TITLE
Add Host To JMX Metrics

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
JMX metrics require having host just like cpu metrics.

# Description of changes
Add host as a metric to jmx metrics unless omit_hostname is true

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests

```
{
  "agent": {
    "debug": true
  },
    "namespace": "HostnameTest",
    "metrics_collected": {
      "cpu": {
        "resources": [
          "*"
        ],
        "measurement": [
          {"name": "cpu_usage_idle", "rename": "CPU_USAGE_IDLE", "unit": "Percent"},
          {"name": "cpu_usage_nice", "unit": "Percent"},
          "cpu_usage_guest"
        ],
        "totalcpu": false,
        "drop_original_metrics": [ "cpu_usage_guest" ],
        "metrics_collection_interval": 10
      },
      "jmx": [
        {
          "endpoint": "localhost:2020",
          "jvm": []
        }
      ]
    }
  }
}
```

Screenshot 2024-05-21 at 12 57 45 PM

```
{
  "agent": {
    "debug": true,
    "omit_empty": true
  },
  "metrics": {
    "append_dimensions": {
      "ImageId": "${aws:ImageId}",
      "InstanceId": "${aws:InstanceId}",
      "InstanceType": "${aws:InstanceType}",
      "AutoScalingGroupName": "${aws:AutoScalingGroupName}"
    },
    "namespace": "HostnameTest",
    "metrics_collected": {
      "cpu": {
        "resources": [
          "*"
        ],
        "measurement": [
          {"name": "cpu_usage_idle", "rename": "CPU_USAGE_IDLE", "unit": "Percent"},
          {"name": "cpu_usage_nice", "unit": "Percent"},
          "cpu_usage_guest"
        ],
        "totalcpu": false,
        "drop_original_metrics": [ "cpu_usage_guest" ],
        "metrics_collection_interval": 10
      },
      "jmx": [
        {
          "endpoint": "localhost:2020",
          "jvm": []
        }
      ]
    }
  }
}
```

Screenshot 2024-05-21 at 1 08 03 PM

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: translator/tocwconfig/sampleConfig/complete_linux_config.yaml [hunk 1]
```diff
@@ -145,6 +145,13 @@ processors:
               from_attribute: ""
               from_context: ""
               pattern: "telemetry.sdk.*"
+            - action: delete
```
Comment (sethAmazon) [inline translator/tocwconfig/sampleConfig/complete_linux_config.yaml:148]: This should be here. I guess the other pr never added it. Now the unit tests pass.

File: translator/translate/otel/receiver/jmx/translator.go [hunk 2]
```diff
@@ -18,12 +18,12 @@ import (
 
 	"github.com/aws/amazon-cloudwatch-agent/internal/util/collections"
 	"github.com/aws/amazon-cloudwatch-agent/tool/paths"
+	"github.com/aws/amazon-cloudwatch-agent/translator/context"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
 )
 
 const (
 	jarPathKey            = "jar_path"
-	targetSystemKey       = "target_system"
```
Comment (sethAmazon) [inline translator/translate/otel/receiver/jmx/translator.go:26]: This is an unused const

File: translator/translate/otel/receiver/jmx/translator_test.go [hunk 3]
```diff
@@ -60,8 +65,11 @@ func TestTranslator(t *testing.T) {
 				require.NotNil(t, got)
 				gotCfg, ok := got.(*jmxreceiver.Config)
 				require.True(t, ok)
-				wantCfg := factory.CreateDefaultConfig()
+				wantCfg := factory.CreateDefaultConfig().(*jmxreceiver.Config)
 				require.NoError(t, component.UnmarshalConfig(testCase.want, wantCfg))
+				if wantCfg.ResourceAttributes != nil && wantCfg.ResourceAttributes[hostnameTag] == hostnameTag {
```
Comment (sethAmazon) [inline translator/translate/otel/receiver/jmx/translator_test.go:70]: Hostname must be the same by the translator hostname. Hostname can only be known at runtime and can't be static

File: translator/translate/otel/receiver/jmx/translator.go [hunk 4]
```diff
@@ -183,6 +186,15 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 		}
 	}
 
+	if !context.CurrentContext().GetOmitHostname() && !conf.IsSet(ec2taggerprocessor.Ec2taggerKey) {
```
Comment (sethAmazon) [inline translator/translate/otel/receiver/jmx/translator.go:189]: Unsure why ec2 tagger was not deleting the host dimensions for jmx metrics. So I manually do not add host when ec2 tagger is added to config

File: translator/translate/otel/receiver/jmx/translator_test.go (hunk 3 above)
Comment (jefchien) [reply translator/translate/otel/receiver/jmx/translator_test.go:70]: Can stub out `os.Hostname` in the translator.

**Review by jefchien — APPROVED** [review]

**Pull request merged**